src/builder/guard.rs

```diff
@@ -100,6 +104,31 @@ impl<'a> Guard<'a> {
                 "Error creating dir: {err}");
         }
 
+        if !self.ctx.container_config.remove_all_except.is_empty() {
+            let root = Path::new("/vagga/root");
+            let exclude_paths = PROTECTED_DIRS.iter()
+                .map(|d| PathBuf::from(d))
+                .chain(self.ctx.container_config.remove_all_except
+                    .iter()
+                    // We validate exclude paths as absolute
+                    .map(|p| PathBuf::from(p.strip_prefix("/").unwrap())))
+                .collect::<Vec<_>>();
+            let mut keep_rel_paths = HashMap::new();
+            for exclude_path in &exclude_paths {
+                for (i, p) in exclude_path
+                    .iter_self_and_parents().enumerate()
+                {
+                    if i == 0 {
+                        keep_rel_paths.insert(p, true);
+                    } else if !keep_rel_paths.contains_key(p) {
```

Comment:
Shouldn't the duplicate key here produce a warning? (in both cases of the if statement). I mean it's something like `[/var, /var/lib]`, which is mistake most of the time, unless you're trying to merge multiple lists which i don't think is a common case.